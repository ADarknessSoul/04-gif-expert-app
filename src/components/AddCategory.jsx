import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {
  
    const [inputValue, setInputValue] = useState('');
  
    const onInputChange = ({target}) => {

        // console.log(target.value);

        setInputValue( target.value );

    }

    const onSubmit = (event) => {

        event.preventDefault();
        const newInputValue = inputValue.trim();
        // console.log(inputValue);

        if(newInputValue.length < 1) return;

        // setCategories(oldArray => [ inputValue, ...oldArray ]);
        onNewCategory(newInputValue);
        setInputValue('');

    }

    return (

        <form onSubmit={ onSubmit }>

            <input 
                type="text" 
                placeholder="Buscar gifs" 
                value={ inputValue } 
                onChange={onInputChange}
            />

        </form>

  )
}
