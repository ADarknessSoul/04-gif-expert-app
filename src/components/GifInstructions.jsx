
export const GifInstructions = () => {


  return (
    
    <>
    
    <section className="contenedor instrucciones">

        <h1 className="instrucciones__titulo">Instrucciones de uso</h1>
        <p>¡Bienvenido a <span>GifExpertApp!</span> Para tener una excelente experiencia se recomienda leer los siguientes pasos:</p>
        <ol>
            <li>Busca tus Gifs en la barra de busqueda</li>
            <li>Utiliza los botones +1 y -1 para agregar y quitar Gifs a tu preferencia</li>
            <li>Utiliza el título de las etiquetas para moverte entre secciones de Gifs</li>
            <li>Borra secciones de Gifs con la X</li>
            <li>¡Diviértete!</li>
        </ol>
        
    </section>

    </>

  )

}
