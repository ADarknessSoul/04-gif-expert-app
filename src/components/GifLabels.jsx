import PropTypes from 'prop-types';

export const GifLabels = ({label, onClickX}) => {

  /*Se llama a la función de callback y le envía el título de la etiqueta */
  const deleteLabel = () => {

    const labelLowerCase = label.toLowerCase();
    onClickX(labelLowerCase);

  };

  /*Se busca el id del título para compararlo con la etiqueta y desplazar el scroll hasta la sección correspondiente */
  const positionScreen = (event) => {

    event.preventDefault();

    const label = event.target.innerHTML;

    const labelWithoutSpaces = label.replaceAll(' ', '');
    
    const h3 = document.querySelector('#' + labelWithoutSpaces);

    h3.scrollIntoView({behavior: "smooth"});

  }

  return (

    <>

      {/* Etiquetas */}

      <div className="historial--redondear">

          <div className="contenido-label">

              <p onClick={ event => positionScreen(event) }>{label}</p>
              <img onClick={ deleteLabel }  src="src\images\x-symbol.svg" alt="símbolo de x" />

          </div>

      </div>

    </>

  )
}

GifLabels.propTypes = {

  label: PropTypes.string.isRequired

}

