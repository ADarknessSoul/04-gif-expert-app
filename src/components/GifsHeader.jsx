import { useEffect } from "react";
import { useState } from "react"
import PropTypes from 'prop-types';

export const GifsHeader = ({category, onClickButton}) => {

    const [limit, setLimit] = useState(3);

      /*useEffect para esperar el resultado asíncrono del useState*/
      useEffect(() => {
        
         onClickButton(limit);

      }, [limit])
      
      /*Se actualiza el state con un valor incrementado del límite */
     const maxLimitAdd = () => {

      if(limit <= 8) {

         setLimit( (c) => c + 1);

      }
         
     }

     /*Se actualiza el state con un valor decrementado del límite */
     const maxLimitDecrease = () => {

         if(limit >= 2) {
            setLimit( (c) => c - 1);
         }

     }


  return (
    <>


         {/* Header del Gif */}
        <button onClick={maxLimitDecrease} className="button button-red">-1</button>
        <h3 id={category.replace(/[^a-z0-9]/gmi, "").replace(/\s+/g, "").replace(' ', '')} aria-label="heading3">{category}</h3> 
        <button onClick={maxLimitAdd} className="button button-green">+1</button>

    </>
  )
}

GifsHeader.propTypes = {

   category: PropTypes.string.isRequired

}