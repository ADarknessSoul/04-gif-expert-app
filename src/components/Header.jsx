import PropTypes from 'prop-types';

export const Header = ({title}) => {

  return (
    
    <>

        <header className="header">

        <div className="contenido-header">

            <img src="\images\logo2.gif" alt="Logo" />

            <h1>{title}</h1>

        </div>



        </header>    
    
    </>

  )
}

Header.propTypes = {

  title: PropTypes.string.isRequired

}

Header.defaultProps = {

  title: "GifExpertApp"

}
