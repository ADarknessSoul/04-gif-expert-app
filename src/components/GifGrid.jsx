import { useState, useEffect } from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import { GifsHeader } from './GifsHeader';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

  const [newLimit, setNewLimit] = useState(3);

  const { images, isLoading } = useFetchGifs(category, newLimit);

  /*Se asigna el nuevo valor del limite al momento de extraer los gifs*/
  const onNewFetch = (limit) => {

    setNewLimit(limit);

  }

  return (

    <>

    <div className='gifs sombra'>

      {/* Título y botones de los Gifs */}
      <section className='gifs__cabecera'>

        <GifsHeader category={category} onClickButton = {value => onNewFetch(value) }/>

      </section>      
      
      {
        
        isLoading && (<h2>Cargando...</h2>) //Ternario simplificado
        
      }

      {/* Diseño de gifs */}
      <section className='card-grid'>
        { 

          images.map( ( image ) => (
            <GifGridItem
              key={image.id}
              { ...image } 
             />
          ))
          
        }
      </section>

    </div>

    </>
  )

}

GifGrid.propTypes = {

  category: PropTypes.string.isRequired

}