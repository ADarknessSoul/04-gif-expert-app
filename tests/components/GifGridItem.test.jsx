import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';

const title = 'Soy un título';
const url = 'https://';

describe('Pruebas en el GrifGridItem', ()=> {

    test('Debe coincidir con el snapshot', ()=> {

        const { container } = render(<GifGridItem title={title} url={url}/>)
         
        expect( container ).toMatchSnapshot();

    }); 
    
    test('Debe mostrar la imagen con el URL y el ATL', ()=> {
    
        render(<GifGridItem title={title} url={url}/>);
         screen.debug();
        // expect( screen.getByRole('img').src ).toBe(url);
        // expect( screen.getByRole('img').alt ).toBe(title);

        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    });

    test('El título se debe mostrar en un párrafo', ()=> {
    
        render(<GifGridItem title={title} url={url}/>);
        expect( screen.getByText(title)).toBeTruthy();
    
    });

});