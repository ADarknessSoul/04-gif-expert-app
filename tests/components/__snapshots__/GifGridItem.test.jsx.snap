// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GrifGridItem Debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Soy un título"
      src="https://"
    />
    <p>
      Soy un título
    </p>
  </div>
</div>
`;
